googlebus/googlebus.go

```diff
@@ -189,12 +190,6 @@ func (m *GoogleBus) Subscribe(ctx context.Context, topicID string, optSubcriptio
 	sctx, cancel := context.WithCancel(ctx)
 
 	sub.unsubscribe = func() {
-		select {
-		case <-sub.done:
-		default:
-			close(sub.done)
-		}
-
```

Comment:
This looks good to me.

`defer close(sub.done)` is called in the below goroutine; and we only `cancel()` its context here.

Reply:
btw

the intention of this code..

```
select {
case <-sub.done: // if sub.done channel is closed, this will immediately return
default:
   // this will only be called if sub.done is not closed
   close(sub.done)
}
```

that is the intention.. as the selecting on a channel with a default, is a trick to make the selecting be non-blocking, but I suppose there is some concurrency issue.. or?

Reply:
David had another version, which worked on already-closed channel:
```
select {
case _, ok := <-sub.done:
    if ok {
        close(sub.done)
    }
default:
}
```

but I don't think it's necessary, as the defer close(sub.done) is always called from the goroutine